lib/s3.js

```diff
@@ -29,20 +29,21 @@ S3.uploadZip = async function (pathToZip) {
   // Call s3.upload and return it as a promise
   // (NOTE: B.promisify doesn't work for s3.upload, it causes s3/managed_upload.js to throw an error `self.service.constructor.__super__ is not a constructor`)
```

Comment:
i see you saying `B.promisify` didn't work, but did you remember to bind it to the `s3` object? I.e., like:

```
B.promisify(s3.upload, {context: s3})
```

?

Reply:
nevermind, even if this works, the idea in general won't work, because `s3.upload` _both_ returns a value _and_ triggers a callback. this means it's not a standard node-style callback method and so we won't be able to use it correctly with `B.promisify`.

Reply:
I just tried that and it gave me the error

```
TypeError: service.getSignatureVersion is not a function
```

Also, promisifying makes it return a promise instead of the uploader, which we need to monitor upload progress.